File: mycollab-web/src/main/java/com/esofthead/mycollab/module/project/view/parameters/MessageScreenData.java
package com.esofthead.mycollab.module.project.view.parameters;

import com.esofthead.mycollab.module.project.domain.criteria.MessageSearchCriteria;
import com.esofthead.mycollab.vaadin.mvp.ScreenData;

public class MessageScreenData {
	public static class Read extends ScreenData<Integer> {

		public Read(Integer params) {
			super(params);
		}
	}

	public static class Search extends ScreenData<MessageSearchCriteria> {

		public Search(MessageSearchCriteria params) {
			super(params);
		}
	}
}

File: mycollab-web/src/main/scala/com.esofthead.mycollab.module.project.view.parameters/MessageScreenData.scala
package com.esofthead.mycollab.module.project.view.parameters

import com.esofthead.mycollab.module.project.domain.criteria.MessageSearchCriteria
import com.esofthead.mycollab.vaadin.mvp.ScreenData

/**
 * @author MyCollab Ltd.
 * @since 5.0.3
 */
object MessageScreenData {

  class Read(params:Integer) extends ScreenData[Integer](params){}

  class Search(params:MessageSearchCriteria) extends ScreenData[MessageSearchCriteria](params){}
}
